docs/staking-and-delegation/managing-stake-sdk.md

```diff
@@ -93,57 +93,133 @@ print(netuids)
+        potential_metric = getattr(hyperparams, 'difficulty', None)  # Adjust based on available params
+
+        if potential_metric is not None:
+            mining_subnets.append({'netuid': subnet.netuid, 'metric': potential_metric})
+            print(f"✅ Using '{potential_metric}' as the relevant metric for subnet {subnet.netuid}.")
+        else:
+            print(f"❌ No valid metric found for subnet {subnet.netuid}. Skipping.")
+    else:
+        print(f"❌ Failed to retrieve hyperparameters for subnet {subnet.netuid}.")
+
+# Step 2: Sort subnets by 'incentive' in descending order and pick top 3
+top_subnets = sorted(mining_subnets, key=lambda x: x['metric'], reverse=True)[:3]
+print(f"Top {len(top_subnets)} subnets selected.")
+
+
+
+# Step 3: Retrieve metagraph and find the top 3 validators by stake and print
+
+# Define staking amount in TAO (ensure it's above the minimum requirement)
+amount_to_stake = 1  # Adjust this if needed
+
+# Convert the amount to the Balance type
+amount_to_stake = bt.Balance.from_tao(amount_to_stake)
+
+# ✅ Declare dictionary before using it
+top_validators_per_subnet = {}  # Stores top validators for each subnet
+
+for subnet in top_subnets:
+    netuid = subnet['netuid']
+    print(f"\n🔍 Fetching metagraph for subnet {netuid}...")
+    
+    start_time = time.time()
+    metagraph = subtensor.metagraph(netuid)
+    print(f"✅ Retrieved metagraph for subnet {netuid} in {time.time() - start_time:.2f} seconds.")
+
+    # Extract validators and their stake amounts
+    uid_stake_pairs = [(uid, metagraph.S[uid]) for uid in range(len(metagraph.S))]
+    
+    # Sort validators by stake in descending order
+    top_validators = sorted(uid_stake_pairs, key=lambda x: x[1], reverse=True)[:3]
+
+    # ✅ Store the metagraph and top validators for Step 4
+    top_validators_per_subnet[netuid] = {
+        "metagraph": metagraph,
+        "validators": top_validators
+    }
+
+    # ✅ Print the top 3 validators for this subnet
+    print(f"\n🏆 Top 3 Validators for Subnet {netuid}:")
+    for rank, (uid, stake) in enumerate(top_validators, start=1):
+        print(f"  {rank}. Validator UID {uid} - Stake: {stake}")
+
+# ✅ Step 4: Stake to each top 3 validators in each top 3 subnets
+for netuid, data in top_validators_per_subnet.items():
+    metagraph = data["metagraph"]
+    top_validators = data["validators"]
+
+    for uid, stake in top_validators:
+        hotkey_ss58 = metagraph.hotkeys[uid]
+
+        print(f"💰 Staking {amount_to_stake} TAO to {hotkey_ss58} on subnet {netuid}...")
+        start_time = time.time()
+        try:
+            # ✅ Use the correct staking amount
+            subtensor.add_stake(wallet=wallet, netuid=netuid, hotkey_ss58=hotkey_ss58, amount=amount_to_stake)
+            print(f"✅ Successfully staked {amount_to_stake} TAO to {hotkey_ss58} on subnet {netuid} in {time.time() - start_time:.2f} seconds.")
```

Comment:
this was telling me i successfully staked when i didn't.
you should not generally put a success message in a try block.
looks like add_stake returns a bool that tells if you it was successful, so success message should be contingent on that I think. I'll take a crack at some updates to the script.
https://docs.bittensor.com/python-api/autoapi/bittensor/core/subtensor/index.html#module-contents